internal/host/host.go

```diff
@@ -514,6 +516,57 @@ func (m *Manager) UpdateNTP(ctx context.Context, h *models.Host, ntpSources []*m
 	return db.Model(h).Update("ntp_sources", string(bytes)).Error
 }
 
+func (m *Manager) UpdateImageStatus(ctx context.Context, h *models.Host, newImageStatus *models.ContainerImageAvailability, db *gorm.DB) error {
+	var hostImageStatuses []*models.ContainerImageAvailability
+	imageFound := false
+
+	if h.ImagesStatus != "" {
+		err := json.Unmarshal([]byte(h.ImagesStatus), &hostImageStatuses)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to unmarshal image statuses for host %s", h.ID.String())
+		}
+	}
+
+	// Check if the image status already exist
+	for index := range hostImageStatuses {
+		if hostImageStatuses[index].Name != newImageStatus.Name {
+			continue
+		}
+
+		// Same result - Nothing to update
+		if hostImageStatuses[index].Result == newImageStatus.Result {
+			m.log.Infof("Image status %s for host %s received same result of %s", newImageStatus.Name, h.ID.String(), newImageStatus.Result)
```

Comment:
if it's expected to happen a lot it will span the log no?

Reply:
Yes, in the best scenario it would happen for every interval.
Do you think it should be removed?

Reply:
Yes! just show changes.